src/wpt/harness.ts

```diff
@@ -219,15 +427,72 @@ globalThis.subsetTestByKey = (
   return testType(testCallback, testMessage);
 };
 
-globalThis.promise_test = (func, name, _properties): void => {
+globalThis.promise_test = (func, name, properties): void => {
   if (!shouldRunTest(name)) {
     return;
   }
 
-  try {
-    globalThis.promises[name] = func.call(this);
-  } catch (err) {
-    globalThis.errors.push(new AggregateError([err], name));
+  const testCase = new Test(name, properties);
+  const promise = testCase.step(func, testCase, testCase);
+
+  if (testCase.error) {
+    globalThis.errors.push(testCase.error);
+  }
```

Comment:
How can a test you just created can have an error property?

Reply:
The function passed to `promise_test` needs to return a `Promise` but is not required to be an `async function`, so if some exception is thrown before it returns the Promise that exception would be caught immediately in `step`.

Reply:
Can you add this as a comment?